admin-dev/themes/new-theme/js/pages/product/components/categories/categories-manager.ts

```diff
@@ -0,0 +1,152 @@
+    this.listenCategoryChanges();
+    this.listenDefaultCategorySelect();
+    this.initCategoryTreeModal();
+  }
+
+  private initCategoryTreeModal(): void {
+    this.addCategoriesBtn.addEventListener('click', () => this.categoryTreeSelector.showModal(
+      this.collectCategories(),
+    ));
+    this.eventEmitter.on(ProductEventMap.categories.applyCategoryTreeChanges, (eventData) => {
+      this.tags.render(eventData.categories);
+    });
+  }
+
+  private collectCategories(): Array<Category> {
+    // these are at first rendered on page load and later updated dynamically
+    const tagsContainer = this.categoriesContainer.querySelector(ProductCategoryMap.tagsContainer) as HTMLElement;
+    const tags = tagsContainer.querySelectorAll(ProductCategoryMap.tagItem);
+    const categories: Array<{ id: number; name: string; isDefault: boolean; }> = [];
+
+    tags.forEach((tag: Element) => {
+      if (tag instanceof HTMLElement) {
+        const defaultCategoryCheckbox = tag.querySelector(ProductCategoryMap.defaultCategoryCheckbox) as HTMLInputElement;
+
+        categories.push({
+          id: Number(tag.dataset.id),
+          name: this.extractCategoryName(tag as HTMLElement),
+          isDefault: defaultCategoryCheckbox.checked,
+        });
+      }
+    });
+
+    return categories;
+  }
+
+  private extractCategoryName(tag: HTMLElement): string {
+    const tagNameElement = tag.querySelector(ProductCategoryMap.categoryNamePreview) as HTMLElement;
+
+    if (tagNameElement) {
+      return tagNameElement.innerText;
+    }
+
+    return '';
+  }
+
+  private renderDefaultCategorySelection(): void {
+    const categories = this.collectCategories();
+    const selectContainer = this.categoriesContainer.querySelector(ProductCategoryMap.defaultCategorySelectContainer) as HTMLElement;
+    const selectElement = this.categoriesContainer.querySelector(ProductCategoryMap.defaultCategorySelectInput) as HTMLElement;
+    selectElement.innerHTML = '';
+
+    categories.forEach((category) => {
+      const optionElement = document.createElement('option');
+      optionElement.value = String(category.id);
+      optionElement.innerHTML = category.name;
+      optionElement.selected = category.isDefault;
+
+      selectElement.append(optionElement);
+    });
+
+    selectContainer.classList.remove('d-none');
+  }
+
+  private listenDefaultCategorySelect(): void {
+    const defaultCategoryInput = this.categoriesContainer.querySelector(ProductCategoryMap.defaultCategorySelectInput) as HTMLInputElement
+    defaultCategoryInput.addEventListener('change', (e) => {
+        const currentTarget = e.currentTarget as HTMLInputElement;
+        const newDefaultCategoryId = Number(currentTarget.value);
+        const categories = this.collectCategories();
+        categories.forEach((category) => {
+          category.isDefault = category.id === newDefaultCategoryId;
+        });
+        this.tags.render(categories);
```

Comment:
Be careful there might be an infinite loop related to events because:
- the `change` event updates the rendered tags
- `Tags.render` emits `categoriesUpdated` at the end
- `categoriesUpdated` triggers `renderDefaultCategorySelection`
- `renderDefaultCategorySelection` rebuilds the select element, so if it triggers a `change` event it will loop